locker_test.go

```diff
@@ -0,0 +1,68 @@
+package gocron
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	testcontainersredis "github.com/testcontainers/testcontainers-go/modules/redis"
+)
+
+func TestEnableDistributedLocking(t *testing.T) {
+	ctx := context.Background()
+	redisContainer, err := testcontainersredis.RunContainer(ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		if err := redisContainer.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	uri, err := redisContainer.ConnectionString(ctx)
+	require.NoError(t, err)
+
+	resultChan := make(chan int, 10)
+	f := func(schedulerInstance int) {
+		resultChan <- schedulerInstance
```

Comment:
```suggestion
		resultChan <- schedulerInstance
		time.Sleep(100 * time.Millisecond)
```

Reply:
It runs fast enough on my Mac that the jobs all attempt to get locks at the same time. I’m guessing that within actions, the jobs are lagging slightly apart and so adding the sleep here causes the lock to stay more